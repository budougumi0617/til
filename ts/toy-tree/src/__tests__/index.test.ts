const { main } = require('../');

describe('toy-tree', () => {
  let stdout;
  let stderr;
  const exec = (argv) => main(argv, stdout, stderr);

  beforeEach(() => {
    stdout = jest.fn();
    stderr = jest.fn();
  });

  test('fixtures', () => {
    const code = exec(['fixtures']);

    expect(stdout.mock.calls[0][0]).toMatchSnapshot();
    expect(code).toBe(0);
  });

  test('-L 2', () => {
    const code = exec(['fixtures', '-L 2']);

    expect(stdout.mock.calls[0][0]).toMatchSnapshot();
    expect(code).toBe(0);
  });

  test('default directory', () => {
    const code = exec([]);

    expect(stdout.mock.calls[0][0].split('\n')[0]).toBe('.');
    expect(code).toBe(0);
  });

  test('invalid directory', () => {
    const dir = '__INVALID__DIRECTORY__';
    const code = exec([dir]);

    expect(stderr.mock.calls[0][0]).toContain(`Error: "${dir}" does not exist`);
    expect(code).toBe(1);
  });

  test('invalid level option', () => {
    const code = exec(['-L 0']);

    expect(stderr.mock.calls[0][0]).toContain('Error: Invalid level');
    expect(code).toBe(1);
  });
});