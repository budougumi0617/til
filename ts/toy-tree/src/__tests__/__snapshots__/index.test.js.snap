// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toy-tree -L 2 1`] = `
"fixtures
|-- dir1
|  |-- file1.txt
|  |-- file2.txt
|  \`-- nest1
|-- dir2
|  |-- file1.txt
|  |-- file2.txt
|  \`-- file3.txt
|-- file1.txt
\`-- file2.txt
"
`;

exports[`toy-tree fixtures 1`] = `
"fixtures
|-- dir1
|  |-- file1.txt
|  |-- file2.txt
|  \`-- nest1
|    |-- deep1.txt
|    \`-- deep2.txt
|-- dir2
|  |-- file1.txt
|  |-- file2.txt
|  \`-- file3.txt
|-- file1.txt
\`-- file2.txt
"
`;
